### ISSUE TITLE
Warnings to get rid of

### ISSUE THREAD

**Issue by rgbkrk** [nteract/archived-desktop-app#1707]
```
WARNING in ./~/spawn-sync/lib/spawn-sync.js
Module not found: Error: Can't resolve 'try-thread-sleep' in '/Users/kylek/code/src/github.com/nteract/nteract/node_modules/spawn-sync/lib'
 @ ./~/spawn-sync/lib/spawn-sync.js 10:10-37
 @ ./~/spawn-sync/index.js
 @ ./~/sync-request/index.js
 @ ./~/datalib/src/import/load.js
 @ ./~/datalib/src/index.js
 @ ./~/vega/src/transforms/Parameter.js
 @ ./~/vega/index.js
 @ ./~/vega-embed/src/embed.js
 @ ./packages/transform-vega/lib/index.js
 @ ./packages/transforms-full/lib/index.js
 @ ./src/notebook/components/notebook.js
 @ ./src/notebook/index.js

WARNING in ./~/encoding/lib/iconv-loader.js
9:12-34 Critical dependency: the request of a dependency is an expression

WARNING in ./~/ajv/lib/async.js
96:20-33 Critical dependency: the request of a dependency is an expression

WARNING in ./~/ajv/lib/async.js
119:15-28 Critical dependency: the request of a dependency is an expression

WARNING in ./~/ajv/lib/compile/index.js
13:21-34 Critical dependency: the request of a dependency is an expression
```

The first one comes from vega, I'm not sure what the dep is for.

The second is from our jsonschema validator, `ajv`.

**Comment by rgbkrk** [comment]
These warnings are still here, mostly just annoying.

**Comment by rgbkrk** [comment]
We should totally figure this one out, I don't like people having to read warnings and wondering if it's the thing that is breaking their setup (when it's usually not).

**Comment by BenRussert** [comment]
I looked into the warning below and think it is related to this closed [cross-sync issue](https://github.com/moxystudio/node-cross-spawn/issues/87)

```
nteract:     WARNING in /home/ben/Documents/nteract/node_modules/cross-spawn/index.js
nteract:     Module not found: Error: Can't resolve 'spawn-sync' in '/home/ben/Documents/nteract/node_modules/cross-spawn'
nteract:      @ /home/ben/Documents/nteract/node_modules/cross-spawn/index.js
...
```

**Comment by rgbkrk** [comment]
Here's all the packages that depend on `cross-spawn` in our monorepo:

```
$ npm ls cross-spawn
nteract-monorepo@0.0.0 /Users/kylek/code/src/github.com/nteract/nteract
├─┬ cross-env@5.1.6
│ └── cross-spawn@5.1.0
├─┬ eslint@4.19.1
│ └── cross-spawn@5.1.0  deduped
├─┬ lerna@2.11.0
│ └─┬ execa@0.8.0
│   └── cross-spawn@5.1.0  deduped
├─┬ lint-staged@7.1.3
│ └─┬ execa@0.9.0
│   └── cross-spawn@5.1.0  deduped
├─┬ next@5.1.0
│ └── cross-spawn@5.1.0  deduped
├─┬ react-styleguidist@7.0.17
│ └─┬ react-dev-utils@5.0.1
│   └── cross-spawn@5.1.0  deduped
├─┬ shell-env@1.0.0
│ └─┬ execa@0.8.0
│   └── cross-spawn@5.1.0  deduped
├─┬ webpack-cli@2.1.5
│ ├── cross-spawn@6.0.5
│ ├─┬ yeoman-environment@2.2.0
│ │ └── cross-spawn@6.0.5
│ └─┬ yeoman-generator@2.0.5
│   └── cross-spawn@6.0.5
└─┬ yargs@11.0.0
  └─┬ os-locale@2.1.0
    └─┬ execa@0.7.0
      └── cross-spawn@5.1.0  deduped
```

**Comment by rgbkrk** [comment]
The warnings come from within webpack land -- perhaps what we need to figure out is why this dep is being webpacked since we don't want this on the frontend anyways. Actually... I think I know... It's in the main process code within `prepare-env.js`

**Comment by rgbkrk** [comment]
It looks like this is [addressed in cross-spawn@6.0.0](https://github.com/moxystudio/node-cross-spawn/blob/master/CHANGELOG.md#600-2018-01-23)

Probably the most relevant PR I made for this, since it's for `shell-env`: https://github.com/sindresorhus/shell-env/pull/7

**Comment by lgeiger** [comment]
These warnings are gone since the latest vega upgrades.
